### PR TITLE
Make BaseManager.register a classmethod (#2356)

### PR DESCRIPTION
**Pull request by Stigjb**
Removes Mypy's complaint about a missing `self` argument when using `register` on a subclass of BaseManager.

### CODE REVIEW

**Comment by Stigjb** [comment]
Fixes #2356. I haven't added a test, as I wasn't sure how or if it's required.

File: stdlib/3/multiprocessing/managers.pyi [hunk 1]
```diff
@@ -20,7 +20,8 @@ _Namespace = Namespace
 class BaseManager(ContextManager[BaseManager]):
     address: Union[str, Tuple[str, int]]
     def connect(self) -> None: ...
-    def register(self, typeid: str, callable: Any = ...) -> None: ...
+    @classmethod
+    def register(cls, typeid: str, callable: Any = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/multiprocessing/managers.pyi]: I noticed that this function has a lot of other parameters in the implementation:
```
    @classmethod
    def register(cls, typeid, callable=None, proxytype=None, exposed=None,
                 method_to_typeid=None, create_method=True):
```
Is it worth adding those too? If they're not important, I can just merge this now and we'll worry about the additional parameters later.

**Comment by Stigjb** [comment]
I tried adding the rest of the arguments, but I'm not completely sure it is correct. I referred to the section on default argument values in PEP 484, and set the default values to `...`, even though I know it is `None` for `callable`, `proxytype`, `exposed` and `method_to_typeid` and `True` for `create_method`. Should the types be `Optional`? There are no complaints even when I pass `None` as an argument.

It seems that it prevented Mypy from complaining about too many arguments or non-existing keyword arguments, though.

**Comment by JelleZijlstra** [comment]
Thanks! Yes, defaults should always be `...` in stubs. If the default is `None`, you should make the type annotation `Optional`.

**Comment by Stigjb** [comment]
I think the PR should be ready now :slightly_smiling_face:

**Pull request merged**